src/main/java/com/bootcamp/demo/controller/AccountController.java

```diff
@@ -26,6 +30,37 @@ public AccountController(Service<Account> service) {
         }
     }
 
+    @PutMapping("/editAccount")
+    ResponseEntity<Account> update(@RequestBody AccountDetails accountDetails) throws ControllerException {
+        try {
+            Account searchedAccount = service.findOne(accountDetails.getUsername());
+
+            if (searchedAccount != null) {
+                Account account = new Account(
+                        accountDetails.getName(),
+                        accountDetails.getEmail(),
+                        accountDetails.getUsername(),
+                        searchedAccount.getPassword(),
+                        accountDetails.getPhoneNumber(),
+                        accountDetails.getAddress(),
+                        accountDetails.getDateOfBirth()
+                );
```

Comment:
We could map an account to accountDetails in the controller and let the service deal with completing the object, like you do with the password.